### PR TITLE
Signin Page in vue app

### LINKED ISSUE #18: signin page in vue app with all functionalities
# I want to be able to sign in app through google, github and local credentials for better user experience.

## Acceptance criteria:
User can sign in through google
User can sign in through github
User can sign in through local credentials
All other routes are protected

### PR DESCRIPTION
**Pull request by AfaqAhmed-XGrid**
## **Issue** #18 
     Title: signin page in vue app

## Description
This pull request sets up the signin page i vue app:
=> Sign in page: Used quasar and custom scss to design sign in page that is fully responsive.
=> Http request: Used axios to make http request to sign in user.
=> Form validation: Added form validation using quasar
=> Protected Routes: Protected all routes except and sign in and sign up.

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

### CODE REVIEW

File: ui_vue/src/views/SignIn.vue [hunk 1]
```diff
@@ -15,6 +15,174 @@ limitations under the License.
 -->
 
 <template>
-  <h1>Sign In Page</h1>
-  <p>Comming soo</p>
```
Comment (saif-xgrid) [inline ui_vue/src/views/SignIn.vue:19]: Please fix this typo

**Review by junaidtariqxgrid — APPROVED** [review]

File: ui_vue/cypress/e2e/signin.cy.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+describe('Sign In', () => {
+
+  it('wrong credentials', () => {
+    cy.visit('/sign-in');
+    cy.url().should('include', '/sign-in');
+    cy.get('[data-cy="email"]', { timeout: 10000 }).type('testuser06@gmail.com');
+    cy.get('[data-cy="password"]').type('Abcd2067*');
+    cy.get('[data-cy="submit"]').click();
+    cy.url().should('include', '/sign-in');
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
```
Comment (XGMqasim) [inline ui_vue/cypress/e2e/signin.cy.ts:12]: Why disabled ESLint?

Comment (AfaqAhmed-XGrid) [reply ui_vue/cypress/e2e/signin.cy.ts:12]: I wanted to add some time between the two tests but eslint was giving error of unnecessary wait.
I added this time gap to observe proper ui changes during testing.

**Review by areeshaxgrid — APPROVED** [review]

File: ui_vue/cypress.config.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ import { defineConfig } from 'cypress';
 export default defineConfig({
   e2e: {
     specPattern: 'cypress/e2e/**/*.{cy,spec}.{js,jsx,ts,tsx}',
-    baseUrl: 'http://localhost:4173'
+    baseUrl: 'http://localhost:5173',
```
Comment (asad-xgrid) [inline ui_vue/cypress.config.ts]: You need to define the base url some where which is included in gitignore eg: .env

File: ui_vue/cypress/support/commands.ts [hunk 4]
```diff
@@ -1,39 +1,87 @@
...
+    });
+  });
+});
+
+Cypress.Commands.add('loginByGithubApi', () => {
+  cy.visit(constants.pages.signIn.link);
+
+  const clientId = import.meta.env.VITE_API_GITHUB_CLIENT_ID;
+  const clientSecret = import.meta.env.VITE_API_GITHUB_CLIENT_SECRET;
+  const code = 'd4674f99a4b3dab1c053';
```
Comment (asad-xgrid) [inline ui_vue/cypress/support/commands.ts]: can you please what code used for and why is not placed in constants or env file

File: ui_vue/cypress/support/commands.ts [hunk 5]
```diff
@@ -1,39 +1,87 @@
...
+    body: {
+      client_id: clientId,
+      client_secret: clientSecret,
+      code: code,
+    },
+    headers: {
+      Accept: 'application/json',
+    },
+  }).then(({ body }) => {
+    const accessToken = body.access_token;
```
Comment (asad-xgrid) [inline ui_vue/cypress/support/commands.ts]: what if the access token is not returned ?

**Review by asad-xgrid — CHANGES_REQUESTED** [review]

File: ui_vue/src/composables/useNotify.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import { useQuasar } from 'quasar';
+
+// Import constants
+import constants from '@/app.constants';
+
+/**
+ * Function to toastify the message on success and failure scenarios while making http requests
+ * @return {Object} returns object of toast functions for success and error
+ */
+export default function useNotify(): {
```
Comment (asad-xgrid) [inline ui_vue/src/composables/useNotify.ts]: Convert it to Arrow function

File: ui_vue/src/styles/colorMode.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.dark-color {
+    color: white;
+}
+
+.dark-bg-color {
+    background-color: #3949AB;
```
Comment (asad-xgrid) [inline ui_vue/src/styles/colorMode.scss]: move colors to the colors class

**Review by asad-xgrid — CHANGES_REQUESTED** [review]

File: ui_vue/cypress.config.ts (hunk 3 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/cypress.config.ts]: Done

File: ui_vue/cypress/support/commands.ts (hunk 4 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/cypress/support/commands.ts]: this is temporary code. I have moved it to env and added comment

File: ui_vue/cypress/support/commands.ts (hunk 5 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/cypress/support/commands.ts]: wrote an assertion there taht would check access_token key in body and if it is not there, test would be failed

File: ui_vue/src/composables/useNotify.ts (hunk 6 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/composables/useNotify.ts]: Done

File: ui_vue/src/styles/colorMode.scss (hunk 7 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/styles/colorMode.scss]: Done

**Review by asad-xgrid — APPROVED** [review]

**Review by saif-xgrid — APPROVED** [review]

**Review by XGMqasim — APPROVED** [review]

File: ui_vue/src/views/SignIn.vue (hunk 1 above)
Comment (AfaqAhmed-XGrid) [reply ui_vue/src/views/SignIn.vue:19]: It is the part of the previous code. Now, it is changed to latest signin page code.

**Pull request merged**